### PR TITLE
[release-4.14] OCPBUGS-21029: Bump library-go and golang.org/x/net

### PR DESCRIPTION
**Pull request by jhadvig**
/assign @stlaz

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,19 +10,19 @@ require (
...
+	github.com/openshift/client-go v0.0.0-20230926161409-848405da69e1
+	github.com/openshift/library-go v0.0.0-20231017173800-126f85ed0cc7
 	github.com/spf13/cobra v1.7.0
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.27.3
-	k8s.io/apiextensions-apiserver v0.27.3
-	k8s.io/apimachinery v0.27.3
-	k8s.io/client-go v0.27.3
-	k8s.io/component-base v0.27.3
+	k8s.io/api v0.28.2
```
Comment (stlaz) [inline go.mod:19]: ocp 4.14 is kube 1.27 (https://access.redhat.com/solutions/4870701)

**Review by stlaz — CHANGES_REQUESTED** [review]

**Comment by jerolimov** [comment]
/retitle [release-4.14] OCPBUGS-21029: Bump library-go and golang.org/x/net

**Pull request merged**